profiles/models.py

```diff
@@ -237,6 +237,58 @@ def import_customer_data(cls, data):
         return result
 
 
+class EncryptedAddress(models.Model, UpdateMixin):
+    street_name = fields.EncryptedCharField(max_length=1024, blank=True)
+    postal_code = fields.EncryptedCharField(max_length=1024, blank=True)
+    city = fields.EncryptedCharField(max_length=1024, blank=True)
+
+    def is_empty(self):
+        return not (self.street_name or self.postal_code or self.city)
+
+
+class VerifiedPersonalInformation(models.Model):
+    profile = models.OneToOneField(
+        Profile, on_delete=models.CASCADE, related_name="verified_personal_information"
+    )
+    first_name = fields.EncryptedCharField(max_length=1024, blank=True)
+    last_name = fields.EncryptedCharField(max_length=1024, blank=True)
+    given_name = fields.EncryptedCharField(max_length=1024, blank=True)
+    national_identification_number = fields.EncryptedCharField(
+        max_length=1024, blank=True
+    )
+    email = fields.EncryptedCharField(max_length=1024, blank=True)
+    municipality_of_residence = fields.EncryptedCharField(max_length=1024, blank=True)
+    municipality_of_residence_number = fields.EncryptedCharField(
+        max_length=4, blank=True
+    )
+    permanent_address = models.OneToOneField(
+        EncryptedAddress,
+        on_delete=models.SET_NULL,
+        null=True,
+        related_name="+",
+        db_index=False,
+    )
+    temporary_address = models.OneToOneField(
+        EncryptedAddress,
+        on_delete=models.SET_NULL,
+        null=True,
+        related_name="+",
+        db_index=False,
+    )
+
+    @classmethod
+    def post_delete(cls, instance, using, **kwargs):
```

Comment:
Check the commit message

Reply:
I read it now, but it doesn't explain why on_delete=models.CASCADE doesn't work.

Reply:
Oh, sorry, the direction would be wrong there.

Reply:
> Oh, sorry, the direction would be wrong there.

Yep, that's right. But look at the data model closely. Is this the right way to do it? Usually I see `OneToOne` fields pointing "the other way". So in here the `OneToOne` field would be in `EncryptedAddress`, pointing to `VerifiedPersonalInformation`. But off course that doesn't work right here, because that wouldn't tell if a given address is the permanent or the temporary address. But is there some other solution for this kind of data?

Reply:
The same thing crossed my mind. But I didn't mentioned it because I realized you wanted to reuse the same class for both fields.

The way to do it i guess would be to make EncryptedAddress a base class and add 2 subclasses with different one-to-one fields and different related_names

Reply:
Your original solution would be much cleaner if not for the extra deletion code.

Reply:
Hmm, that sounds like it could work 🤔

Reply:
Yeah, kinda sad it would obscure the main model by sprinkling its field definitions elsewhere. OTOH maybe that's the Django Way. :)

Reply:
I updated this PR with the way discussed in this thread. I think it is the Django way as Django now takes care of some things that required extra code in the previous version. There is one more model and database table with this approach, but perhaps that's not too bad.